addons/mail/tests/test_mail_template.py

```diff
@@ -159,3 +159,16 @@ def test_mail_template_acl_translation(self):
         # Employee without mail_template_editor group cannot create dynamic translation for mail.render.mixin
         with self.assertRaises(AccessError):
             Translation.with_user(self.user_employee).create(subject_translation_vals)
+
+    def test_server_archived_usage_protection(self):
+        """ Test the protection against using archived server (servers used cannot be archived) """
+        IrMailServer = self.env['ir.mail_server']
+        server = IrMailServer.create({
+            'name': 'Server',
+            'smtp_host': 'archive-test.smtp.local',
+        })
+        self.mail_template.mail_server_id = server.id
+        with self.assertRaises(UserError, msg='Server cannot be archived because it is used'):
+            server.action_archive()
+        self.mail_template.mail_server_id = IrMailServer
+        server.action_archive()  # No more usage -> can be archived
```

Comment:
Could check that server.active is False, to have a complete test :)

Reply:
Yes, of course, changed